### PR TITLE
Remove old demos, fix airbnb_reviews and add store demo

### PR DESCRIPTION
**Pull request by PatrikDurdevic**

### CODE REVIEW

**Comment by PatrikDurdevic** [comment]
Had to remove bz2 dataset compression in the process since pip wasn't packaging it

**Comment by PatrikDurdevic** [comment]
Also, there is only 1 integration test for airbnb reviews since the store integration test takes too long (3 hours, it's currently commented out for that reason)

File: .github/workflows/test_without_dev_deps.yaml [hunk 1]
```diff
@@ -37,26 +37,25 @@ jobs:
...
-          window_size = "2d"
-          minimum_data = "2020-01-22"
-          maximum_data = "2020-03-29"
+          df = trane.datasets.load_airbnb_reviews()
+          entity_col = "location"
+          time_col = "date"
+          window_size = "1m"
+          table_meta = trane.typing.infer_table_meta(df, entity_col, time_col)
+          table_meta["location"] = trane.typing.column_schema.ColumnSchema(
+              trane.typing.logical_types.Categorical, semantic_tags={"category", "primary_key"}
```
Comment (gsheni) [inline .github/workflows/test_without_dev_deps.yaml:46]:
```suggestion
              ('Categorical', {"category", "primary_key"})
```

Comment (gsheni) [reply .github/workflows/test_without_dev_deps.yaml:46]: Optional (trying to make things easier for users)

File: trane/datasets/data/store/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Dell Store 2
```
Comment (gsheni) [inline trane/datasets/data/store/README.md:1]: Add these files to here:
https://github.com/trane-dev/Trane/blob/3f2b256e3acd44c43ab03a7f3cff3e6d4d54893f/pyproject.toml#L81

File: trane/datasets/data/store/LICENSE [hunk 3]
```diff
@@ -0,0 +1,340 @@
+		    GNU GENERAL PUBLIC LICENSE
```
Comment (gsheni) [inline trane/datasets/data/store/LICENSE:1]: Add this file to here:
https://github.com/trane-dev/Trane/blob/main/pyproject.toml#L69

**Review by gsheni — CHANGES_REQUESTED** [review]
Left some minor comments. Overall, looks good.

**Comment by gsheni** [comment]
> Had to remove bz2 dataset compression in the process since pip wasn't packaging it

This is because the file was not included here:
https://github.com/trane-dev/Trane/blob/main/pyproject.toml#L81

**Comment by gsheni** [comment]
> Also, there is only 1 integration test for airbnb reviews since the store integration test takes too long (3 hours, it's currently commented out for that reason)

Yep, this is fine. We should try to rely on unit tests as much as possible.

File: .github/workflows/test_without_dev_deps.yaml (hunk 1 above)
Comment (gsheni) [reply .github/workflows/test_without_dev_deps.yaml]: Disregard this comment.

**Review by gsheni — APPROVED** [review]

**Pull request merged**